### DISCUSSION TITLE
Ring token expires almost immediately and doesn’t update

### DISCUSSION THREAD

**Discussion by Quilgy** [dgreif/ring#907] [General]
Hey there, I’m running Homebridge Ring on Mac OS Catalina and am finding my Ring token is expiring almost straight away (as expected) but it’s not updating and saving the updates in the config. This is resulting in an error connecting to the API as the refresh token is not valid. 

I can renew the token using the email login and 2FA through the Config UI and save it, and it works for a few minutes before getting the error. 

Any idea what might be causing it not to save the refreshed token? 

Versions:
Node.js Version | v14.8.0
Npm v6.14.7
homebridge-ring v9.24.0

**Comment by trackready** [comment]
Same/similar issue. Starting around a month ago I had to refresh my token several times within a week. Seems to stabilize, now i'm on day 2, refreshing the token but it seems to almost immediately expire

**Comment by dgreif** [comment]
I think this issue is specific to old versions of MacOS.  Is it possible for you to update? Or is your hardware not supported on newer versions?

**Reply by Quilgy** [reply]
Ah yeah I'm on Catalina 10.15.7 which is the newest the hardware will support. You reckon that's the reason?

**Reply by dgreif** [reply]
Yeah, that seems to be the common factor for everyone experiencing this issue

**Reply by Quilgy** [reply]
Any idea from which version it's OK? I might see if I can patch upgrade to Big Sur if that would work.

**Reply by dgreif** [reply]
I haven't heard any issues from Big Sur users, so I think that would be safe
